src/main/java/seedu/address/ui/HistoryCommandListPanel.java

```diff
@@ -0,0 +1,49 @@
+package seedu.address.ui;
+
+import java.util.Comparator;
+import java.util.logging.Logger;
+
+import javafx.collections.ObservableList;
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ListView;
+import javafx.scene.layout.Region;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.model.history.HistoryCommand;
+
+/**
+ * Panel containing the list of persons.
+ */
+public class HistoryCommandListPanel extends UiPart<Region> {
+    private static final String FXML = "HistoryCommandListPanel.fxml";
+    private final Logger logger = LogsCenter.getLogger(HistoryCommandListPanel.class);
+
+    @javafx.fxml.FXML
+    private ListView<HistoryCommand> historyCommandListView;
+
+    /**
+     * Creates a {@code HistoryCommandListPanel} with the given {@code ObservableList}.
+     */
+    public HistoryCommandListPanel(ObservableList<HistoryCommand> commandList) {
+        super(FXML);
+        commandList.sort(Comparator.comparingInt(HistoryCommand::getIndex).reversed());
```

Comment:
my understanding is that this doesn't work bc the `ObservableList` won't be modified..? anyway this is only run in the constructor, so when the list gets updated, this sorting will not run